crates/uv-python/src/downloads.rs

```diff
@@ -125,11 +125,40 @@ pub struct PythonDownloadRequest {
     pub(crate) prereleases: Option<bool>,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum ArchRequest {
+    Explicit(Arch),
+    Environment(Arch),
+}
+
+impl Display for ArchRequest {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Explicit(arch) | Self::Environment(arch) => write!(f, "{arch}"),
+        }
+    }
+}
+
+impl ArchRequest {
+    pub(crate) fn satisfied_by(self, arch: Arch) -> bool {
+        match self {
+            Self::Explicit(request) => request == arch,
+            Self::Environment(env) => env.supports(arch),
```

Comment:
Key changes

The first line is new behavior, requiring exact equality.

The second retains the desirable existing behavior.